docs/user-guide: update installation (#135)
nix-community/nixd: docs/user-guide.md

## User Guide

### Installation
This program can be found in [NixOS/nixpkgs](https://github.com/NixOS/nixpkgs), there are different ways to install nixd, pick your favourite:

<details>
<summary>NixOS Configuration</summary>

```nix
{ pkgs, ... }: {
  environment.systemPackages = with pkgs; [
    nixd
  ];
}
```

</details>

<details>
<summary><b>nix-env</b>(legacy commands)</summary>
On NixOS:

```console
$ nix-env -iA nixos.nixd
```

On Non NixOS:

```console
$ nix-env -iA nixpkgs.nixd
```

</details>

<details>

> <summary>With <b>nix profile</b></summary>

Reviewer: <del>With </del>

```console
$ nix profile install github:nixos/nixpkgs#nixd
```

</details>

And our flake.nix provides a package named `nixd`, and an overlay to nixpkgs that add the `nixd` package.

Note that please do NOT override nixpkgs revision for nixd inputs.
The source code have tested on specific version on NixOS/nix, which may not work at your version.

> Version 1.0.0 is very experimental and has some known bugs, it is recommended to compile from source.

Reviewer: Move this sentences *above* the instructions for installing from nixpkgs, many users will choose nixpkgs immediately without reading the doc here.

## User Guide

### Installation

> Version 1.0.0 is very experimental and has some known bugs, it is highly recommended to install nixd from source.

Reviewer: How about just do not mention `1.0.0` because we may release something like `1.0.1` or whatever.

```diff
- Version 1.0.0 is very experimental and has some known bugs, it is highly recommended to install nixd from source.
+ At this time (2023-06-14), nixd is under rapid development and it is highly recommended to install nixd from source.
```

> This program can be found in [NixOS/nixpkgs](https://github.com/NixOS/nixpkgs), there are different ways to install nixd, pick your favourite:

Reviewer: ```suggestion
Package `nixd` can be found in [nixpkgs](https://github.com/NixOS/nixpkgs), there are different ways to install nixd, pick your favourite:
```

## User Guide

> ### Installation

Reviewer: Maybe we should move 

https://github.com/nix-community/nixd/blob/db1a463720d4e5d569b52b0c9af2e6c4736e8608/docs/user-guide.md?plain=1#L49

here to recommend users compile from source. Version 1.0.0 is very experimental and has some known bugs.

> This program can be found in [NixOS/nixpkgs](https://github.com/NixOS/nixpkgs),there are different ways to install nixd, pick your favourite:

Reviewer: ```suggestion
This program can be found in [NixOS/nixpkgs](https://github.com/NixOS/nixpkgs), there are different ways to install nixd, pick your favourite:
```

Author: done

<details>

> <summary>Via configuration.nix in NixOS</summary>

Reviewer: ```suggestion
<summary>NixOS Configuration</summary>
```

Author: done

```nix
{ pkgs, ... }: {
  environment.systemPackages = with pkgs; [
    nixd
  ];
}
```

</details>

<details>

> <summary>With <b>nix-env</b>(legacy commands)</summary>

Reviewer: ```suggestion
<summary><b>nix-env</b>(legacy commands)</summary>
```

Author: done
